xarray/tests/test_dataset.py

```diff
@@ -3505,6 +3505,44 @@ def test_sortby(self):
         actual = ds.sortby(['x', 'y'], ascending=False)
         self.assertDatasetEqual(actual, ds)
 
+    def test_ipython_key_completion(self):
+        ds = create_test_data(seed=1)
+        actual = ds._ipython_key_completions_()
+        expected = ['var1', 'var2', 'var3', 'time', 'dim1',
+                    'dim2', 'dim3', 'numbers']
+        for item in expected:
+            ds[item]
+        self.assertTrue(len(actual) == len(expected))
+        self.assertTrue(set(actual) == set(expected))
```

Comment:
`assert sorted(actual) == sorted(expected)` would be slightly cleaner than these tests